exec/java-exec/src/test/java/org/apache/drill/BaseTestInheritance.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill;
+
+import org.apache.drill.categories.UnlikelyTest;
+import org.apache.drill.test.BaseTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+
+import java.lang.reflect.Method;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertTrue;
+
+public class BaseTestInheritance extends BaseTest {
+
+  @Test
+  @Category(UnlikelyTest.class)
+  public void verifyInheritance() {
+    // Get all BaseTest inheritors
+    Reflections reflections = new Reflections("org.apache.drill", new SubTypesScanner(false));
+    Set<String> baseTestInheritors = reflections.getSubTypesOf(BaseTest.class).stream()
+        .map(Class::getName)
+        .collect(Collectors.toSet());
+    // Get all tests
+    Set<String> testClasses = reflections.getSubTypesOf(Object.class).stream()
+        .filter(c -> !c.isInterface())
+        .filter(c -> c.getSimpleName().toLowerCase().contains("test"))
+        .filter(c -> {
+          for (Method m : c.getDeclaredMethods()) {
+            if (m.getAnnotation(Test.class) != null) {
+              return true;
+            }
+          }
+          return false;
+        })
+        .map(Class::getName)
+        .collect(Collectors.toSet());
+
+    testClasses.removeAll(baseTestInheritors);
+    String illegitimateTests = String.join("\n", testClasses);
+    assertTrue(String.format("The following test classes are not inherited from BaseTest:\n%s", illegitimateTests),
```

Comment:
We may use assert equals and specify empty collection, so it wouldn't be required to append the list into the error string.